osquery/tables/system/darwin/sharing.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+#include "osquery/tables/system/darwin/sharing.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::string kInternetSharingPath = "/Library/Preferences/SystemConfiguration/com.apple.nat.plist";
+const std::string kRemoteAppleManagementPath = "/Library/Application Support/Apple/Remote Desktop/RemoteManagement.launchd";
+const std::string kRemoteBluetoothSharingPath = "/Library/Preferences/ByHost/";
+const std::string kRemoteBluetoothSharingPattern = "com.apple.Bluetooth.%";
+
+
+bool remoteAppleManagementPlistExists() {
+  auto internet_sharing_status = SQL::selectAllFrom("file", "path", EQUALS, kRemoteAppleManagementPath);
+  if (internet_sharing_status.empty()) {
+    return false;
+  }
+  return true;
+}
+
+int getScreenSharingStatus() {
+  Boolean persistent;
+  if (remoteAppleManagementPlistExists()) {
+    return 0;
+  } else {
+    return SMJobIsEnabled(kSMDomainSystemLaunchd, CFSTR("com.apple.screensharing"), &persistent);
+  }
+}
+
+int getRemoteManagementStatus() {
+  return remoteAppleManagementPlistExists() ? 1 : 0;
+}
+
+int getFileSharingStatus() {
+  Boolean persistent;
+  int smbStatus = SMJobIsEnabled(kSMDomainSystemLaunchd, CFSTR("com.apple.smbd"), &persistent);
+  int fileServerStatus = SMJobIsEnabled(kSMDomainSystemLaunchd, CFSTR("com.apple.AppleFileServer"), &persistent);
+  if (smbStatus == 1 || fileServerStatus == 1) {
+    return 1;
+  } else {
```

Comment:
You don't need this else. (I think you could also just `return smbStatus | fileServerStatus;` but I haven't tested it 😛 )